src/main/java/com/conveyal/datatools/manager/models/Deployment.java

```diff
@@ -395,6 +399,33 @@ public String generateBuildConfigAsString() {
         return writeToString(this.parentProject().buildConfig);
     }
 
+    /**
+     * Check if there are active jobs to fetch or process a new version for all feed sources (except for the
+     * feedSourceIdToSkip). This is helpful in the context of auto-deploying to OTP to determine if a given feed is the
+     * last to process a newly fetched version (we only want the deployment to occur if all fetches have completed).
+     * TODO: what if some of the feed fetches loaded new GTFS data that had critical errors? Should we not
+     *  auto-deploy?
+     */
+    public boolean hasFeedFetchesInProgress(String feedSourceIdToSkip) {
+        // Collect this deployment's feed source IDs (from feed versions) to check the active jobs against.
+        Set<String> feedSourceIds = retrieveFullFeedVersions().stream()
+            .map(version -> version.feedSourceId)
+            .filter(id -> !id.equals(feedSourceIdToSkip))
+            .collect(Collectors.toSet());
+        // If there are any active fetch/process feed jobs for one of the feed source IDs, return true.
+        return StatusController.filterActiveJobs(StatusController.getAllJobs()).stream().anyMatch(job -> {
```

Comment:
Generic util functions that are used by multiple classes should be moved to a generic util file. In this case, the `filterActiveJobs` and `getAllJobs` should be extracted into a `JobUtils` file.